cmd/ci-operator/main.go

```diff
@@ -704,10 +708,11 @@ func (o *options) resolveInputs(steps []api.Step) error {
 	o.jobSpec.SetNamespace(o.namespace)
 
 	//If we can resolve the field, use it. If not, don't.
-	if routeGetter, err := routeclientset.NewForConfig(o.clusterConfig); err != nil {
-		log.Printf("could not get route client for cluster config")
+	if client, err := ctrlruntimeclient.New(o.clusterConfig, ctrlruntimeclient.Options{}); err != nil {
```

Comment:
Why not just add all the required schemes in Options here instead of adding schemes to the runtime in `addSchemes` etc.. ?

Reply:
Then I would need to create a new scheme everytime I construct a client, using the global scheme allows me to do it once and be done

Reply:
Also, another concern is why we need to add those schemes in runtime?

Reply:
Because otherwise we don't have it. The typed clients do this in an `init()`, e.G. https://github.com/openshift/ci-tools/blob/76e84167a77c09851dba6de95b46f95ecaeadc60/vendor/github.com/openshift/client-go/template/clientset/versioned/scheme/register.go#L39